dandi/metadata.py

```diff
@@ -176,12 +194,14 @@ def _check_decimal_parts(age_parts):
             age_parts[-1],
             flags=re.I,
         )
+        if m is None:
+            raise ValueError(age_parts)
```

Comment:
```suggestion
            raise ValueError(f"Failed to parse the trailing part of age {age_parts[-1]!r}")
```
or something like that
- other parts have no relation to parsing of the last one
- to provide a hint on what/why the value is wrong

may be even worth placing that regex into a variable and including it in the error message to make it even more informative